File: LeetCode/Study Plan/Practice Problems/Medium/Find The Index Of The First Occurrence In A String/Java/findIndexOfFirstOccurence.java
class Solution {
    public int strStr(String haystack, String needle) {
        int hSize = haystack.length();
        int nSize = needle.length();
        int i=0;

        while(i<hSize) {
            int k = 0;
            int start = i;
        
            while(start<hSize && k<nSize && haystack.charAt(start) == needle.charAt(k)) {
                start++;
                k++;
            }

            if (k == nSize) return i;
            i++;
        }

        return -1;
    }
}

File: LeetCode/Study Plan/Practice Problems/Medium/Find The Index Of The First Occurrence In A String/Go/findIndexOfFirstOccurence.go
package main

import "fmt"

func strStr(haystack string, needle string) int {
	hSize := len(haystack)
	nSize := len(needle)
	i := 0

	for i < hSize {
		start := i
		k := 0
		for start < hSize && k < nSize && haystack[start] == needle[k] {
			start++
			k++
		}

		if k == nSize {
			return i
		}
		i++
	}

	return -1
}

func main() {
	fmt.Println(strStr("sadbutsad", "sad"))
}
